src/infra/database-interface.ts

```diff
@@ -12,6 +11,30 @@ import { TRACKER_START_TIME, suggestion_entry } from "../components/server-sugge
 import { link_blacklist_entry } from "../private-types.js";
 import { moderation_entry } from "../components/moderation/moderation-common.js";
 
+export type database_credentials = {
+    user: string;
+    password: string;
+};
+
+export type database_info = {
```

Comment:
This is best placed in wheatley.ts, as it is the schema for the general bot info. Similarly, all other schemas are in their appropriate component.ts files.